core/src/main/java/org/opensearch/sql/ast/expression/QualifiedName.java

```diff
@@ -25,14 +25,27 @@
 public class QualifiedName extends UnresolvedExpression {
   private final List<String> parts;
 
+  @Getter
+  private final Boolean isMetadataField;
+
   public QualifiedName(String name) {
+    this(name, Boolean.FALSE);
+  }
+
+  public QualifiedName(String name, Boolean isMetadataField) {
     this.parts = Collections.singletonList(name);
+    this.isMetadataField = isMetadataField;
+  }
+
+  public QualifiedName(Iterable<String> parts) {
+    this(parts, Boolean.FALSE);
   }
 
   /**
    * QualifiedName Constructor.
    */
-  public QualifiedName(Iterable<String> parts) {
+  public QualifiedName(Iterable<String> parts, Boolean isMetadataField) {
```

Comment:
Would it be better to determine `isMetadataField` dynamically rather than through constructor parameter?

Reply:
It's easiest to define it at constructor time... since we need to look up the expression tree.